[16.0][MIG] account_reversal (reversal usability improvements)

rousseldenis opened the pull request:
- Adds reversal id field

Commit:
[ADD] port account_reversal from v6.0 and add features : to_be_reversed flag and reversal_id many2one
(lp:c2c-addons/6.1  rev 5609)

Commit:
[MIGR] set installable to False on all the modules, the flag has to be reactivated when the modules are migrated

Commit:
[MIGR] set account_reversal installable

Commit:
[IMP] manifest

Commit:
[FIX] encoding should be coding

Commit:
[FIX] license mess, the whole addon is release under AGPLv3 but a python module contains GPLv2

Commit:
[IMP] python formatting, OpenERP standards

Commit:
[FIX] remove useless type in views

Commit:
[FIX] one more type to remove

Commit:
[IMP] views for version 7.0

yvaucher committed:
[IMP] python formating

Commit:
[IMP] Translation template files for all modules.

Commit:
Launchpad automatic translations update.

Commit:
[FIX] Avoid a crash when the next period doesn't exist (for example : you are in December 2013 and FY 2014 is not created yet).
Add prefer_normal_period in period selection.

Commit:
Launchpad automatic translations update.

nbessi committed:
Set all addons to uninstallable

nbessi committed:
Rename all addons to xxx_unported to be travis compliant

Commit:
Revert "Rename all addons to xxx_unported to be travis compliant"

This reverts commit 7eb182744f3057dd5cb9e35faf230412aa7a2331.

Commit:
[FIX] move all unported addons into __unported__ folder instead of adding _unported suffix

nbessi committed:
[FIX] flake8 PEP8 for module account_reversal

Commit:
Move account_reversal from __unported__ to root folder

Commit:
Move the description in README.rst

Commit:
Set the module installable

Commit:
Use relative imports

Commit:
Size useless on char fields

sbidoul committed:
[FIX] pylint relative imports

sbidoul committed:
[IMP] improve account_reversal fr translation

bringsvor committed:
Rewritten for new api.

Added myself to the readme file.

bringsvor committed:
Missing default value for company_id.

rschnapka committed:
initial german translation (mostly)

rschnapka committed:
removed extra linefeed

gurneyalex committed:
Add OCA as author of OCA addons

In order to get visibility on https://www.odoo.com/apps the OCA board has
decided to add the OCA as author of all the addons maintained as part of the
association.

bringsvor committed:
Added a unittest as requested.

Commit:
[FIX] Nonexisting field in data entry

Commit:
[FIX][account_reversal] To be reversed should be false after reversal

Commit:
Make Warnings translatable

Commit:
Fixup tests

Use current year
Simplify _create_move

sbidoul committed:
[FIX] account_reversal: correctly display the reversal move

This issue was introduced by ac3744cbff4e43db749d194b141dfcd57a8d9460.
Before that commit, to_be_reversed remained true even after reversal.

Commit:
Use api.multi

sbidoul committed:
[FIX] account_reversal: fix test

sbidoul committed:
[IMP] account_reversal: update contributors

sbidoul committed:
[FIX] account_reversal: allow reversing move in closed periods, fixes #265

sbidoul committed:
[FIX] add missing return statement

Commit:
[FIX][account_reversal] When period is set on the reversal wizard. reversal_period_id is never transformed to a recordset

Commit:
[FIX][account_reversal] When period is set on the reversal wizard. reversal_period_id is never transformed to a recordset

Commit:
[IMP][account_reversal] Naming convention

Commit:
[IMP][account_reversal] Naming convention

sbidoul committed:
Revert "[FIX][account_reversal] When period is set on the reversal wizard. reversal_period_id is never transformed to a recordset"

Commit:
[CHG][account_reversal] Modify test according changes in _move_reversal method

yvaucher committed:
Add missing default oca icons

oca-transbot committed:
OCA Transbot updated translations from Transifex

sbidoul committed:
[UPD] prefix versions with 8.0

pedrobaeza committed:
[MIG] Make modules uninstallable

sbidoul committed:
[FIX] remove en.po that was erroneously created by transbot

pedrobaeza committed:
[MIG] Rename manifest files

Commit:
[10.0][MIG] account_reversal migration (#445)

* OCA Transbot updated translations from Transifex
* Add option to reconcile the reversal move with the original move (#387)
* [IMP] account_reversal: Several improvements
* Default date for reverse move is not start_date of the period following the period of the original move (and not the period following today's period)
* Return form view of account.move if only 1 move is reversed
* Add option to reconcile with the reversal entry.
* [10.0][MIG] account_reversal migration
* [CHG] skid Odoo method to allow better performance on huge volume

JordiBForgeFlow committed:
[11.0][MIG] account_reversal migration

etobella committed:
Improve test

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

sbidoul committed:
remove obsolete .pot files [ci skip]

oca-travis committed:
[UPD] Update account_reversal.pot

osoul-dev committed:
Translated using Weblate (Arabic)

Currently translated at 86.7% (26 of 30 strings)

Translation: account-financial-tools-11.0/account-financial-tools-11.0-account_reversal
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-11-0/account-financial-tools-11-0-account_reversal/ar/

oca-travis committed:
[UPD] Update account_reversal.pot

Commit:
[FIX] account_reversal - Reverse multiple journal entries.

Commit:
Translated using Weblate (Dutch)

Currently translated at 100.0% (30 of 30 strings)

Translation: account-financial-tools-11.0/account-financial-tools-11.0-account_reversal
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-11-0/account-financial-tools-11-0-account_reversal/nl/

Commit:
[FIX] account reversal typo

oca-travis committed:
[UPD] Update account_reversal.pot

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

Commit:
[11.0][IMP] Journal Entries to be Reversed menu

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-tools-11.0/account-financial-tools-11.0-account_reversal
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-11-0/account-financial-tools-11-0-account_reversal/

rousseldenis committed:
[IMP] account_reversal: pre-commit stuff

rousseldenis committed:
[16.0][MIG] account_reversal

Migration to 16.0 by expanding core features adding:

- The line reason to reversal
- The 'To Be Reversed' field and filter
- Displaying the move reason not only for invoices

Commit:
[IMP] account_reversal: add reversal_id field

This commit adds the `reversal_id` field in order to improve
the UX of the move reversal functionality. The field has been
added to the form view close to the `reversed_entry_id` field
already present in the view.

This commit also introduces a constraint preventing the user
to flag a move as to be reversed when a non cancelled reversal
move has already been created. This allows simplifying the domain
used on the `to_be_reversed` filter.

Finally, it also convert the deprecated `SavepointCase` test
into a `TransactionCase` one.

Fixes: #1761

IT-Ideas commented on account_reversal/models/account_move.py:
```diff
@@ -0,0 +1,26 @@
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    to_be_reversed = fields.Boolean(
```
Hey @sbejaoui, just a reminder about the PR that will move this field into another new module.
see here https://github.com/OCA/account-financial-tools/pull/1771

IT-Ideas replied:
Oups @rousseldenis , sorry I thought this was @sbejaoui 's PR. Actually @sbidoul asked me to create a PR last Friday it seems that you already made the job 🙂 
@sbidoul should I close my PR? In that case we should close your issue https://github.com/OCA/account-financial-tools/issues/1761 in this PR message.

sbidoul replied:
Ow, my bad, I had not thought this would reappear as a migration.

I guess it makes sens to keep the name?

However, it seems this PR does not have the reversal_id field and navigation.

sbidoul replied:
On the other hand, the account_reversal name is a little bit misleading because the reversal feature is part of Odoo core.

IT-Ideas replied:
Hello @rousseldenis,
Could you just add the `reversal_id field` [field](https://github.com/OCA/account-financial-tools/pull/1771/commits/7f5f9f21355dda215d9f069add6124843228b8c2#diff-656b2aceb1c3e57e4f626cc6a3db29be015da78b9f1979dcb4920b831c44e0ccR18-R23) and associated search and field in form view?
Thanks 🙂 
Thanks!

IT-Ideas commented:
Hello @rousseldenis, I just added a commit in order to fix #1761. Could you add it in the PR message?

sbidoul committed:
[IMP] account_reversal: improve fields ordering in view

After this commit, the order is
- To be reversed flag
- Reversal Entry
- Reversal of

Before, the flag was between the entries which was weird.

sbidoul committed:
[FIX] account_reversal: handle lines with empty label

sbidoul committed:
[IMP] account_reversal: clarify help and documentation

sbidoul commented:
I have pushed a couple of minor fixes and clarifications.

I think it's ok to keep the module name as is to reduce migration burden for users of the module in older Odoo versions.

vdewulf reviewed (approved):
Tested on runboat with success.

- adding a checkbox and filter "to be reversed" on account moves. => OK
- allowing to add a line reason (the move reason already exists in core). => OK
- makes the reason for account entries visible for normal entries too. Not really tested.

The list of Entries to be reversed exists and in the list, the entry disappears once reversed. the info of the reversed entry is visible on both journal entries.

Works as expected.
![Sélection_362](https://github.com/user-attachments/assets/61fcb4f5-f5f6-4404-92d2-cc53151c54da)
![Sélection_360](https://github.com/user-attachments/assets/68efdf70-efaf-4d0e-9075-78d674820e19)

![Sélection_361](https://github.com/user-attachments/assets/b819fa8c-9fa1-4aaa-b293-fd624b2553d0)

vdewulf commented:
/ocabot merge nobump

sbidoul commented:
rebased to investigate the merge failure

lmignon commented:
@sbidoul Can you reopen this one? It's still used on our projects...

sbidoul commented:
So let's merge it?

sbidoul commented:
It's green and works fine in several production instances.

/ocabot merge nobump
